src/request_generator.rs

```diff
@@ -20,7 +22,8 @@ pub struct RequestGenerator {
     pub method: Method,
     pub version: Version,
     pub headers: HeaderMap,
-    pub body: Bytes,
+    pub body: Option<Bytes>,
+    pub body_lines_reader: Option<Arc<Mutex<LineReader>>>,
```

Comment:
Are really multiple threads mutating the reader at the same time?
Otherwise

```suggestion
    pub body_lines_reader: Option<Arc<LineReader>>,
```
should be enough, this seems sequential access only to me.
Without threads even Option<LineReader> should be working fine.